libtransmission/torrent.cc

```diff
@@ -2435,6 +2435,28 @@ size_t tr_torrentFindFileToBuf(tr_torrent const* tor, tr_file_index_t file_num,
     return tr_strvToBuf(tr_torrentFindFile(tor, file_num), buf, buflen);
 }
 
+void tr_torrent::setDownloadDir(std::string_view path, bool isNewTorrent)
+{
+    download_dir = path;
+    markEdited();
+    setDirty();
+    refreshCurrentDir();
+
+    if (isNewTorrent)
```

Comment:
Is this conditional necessary? Is there a reason we couldn't remove `isNewTorrent` and always call lines 2447-2457?

Reply:
There are only two calls to this method: one on new torrents, and one on data location **move**.
It makes sense to be at 100% on new torrents, but not on data location **move**, which case we don't want to affect the current progress.